### DISCUSSION TITLE
Migration from unencrypted to encrypted database file

### DISCUSSION THREAD

**Discussion by damianogiusti** [skolson/KmpSqlencrypt#3] [Q&A]
Hi! Does the library provide a wrapper or a sort of convenience method to perform the migration from an unencrypted to an encrypted database? I'm not yet relying on your library. Still, I'm trying to implement this feature in a Kotlin Multiplatform module of mine, and since I'm finding some issues I was considering using your project as a reference.

Basically, the operations that should be done to migrate a plaintext db to an encrypted one are the following:

```
$ ./sqlcipher plaintext.db
sqlite> ATTACH DATABASE 'encrypted.db' AS encrypted KEY 'newkey';
sqlite> SELECT sqlcipher_export('encrypted');
sqlite> DETACH DATABASE encrypted;
```

Thanks in advance!

**Comment by skolson** [comment]
I haven't used the API to do an ATTACH yet. But I think this would be really easy, as the ATTACH and DETACH are (I believe) both raw SQL commands that should work fine using the existing SqlDatabase execute(...) function. The SELECT would work either as raw SQL or using the query support in the API.  
If we made a wrapper, are you thinking a SqlDatabase function like 
`fun encrypt( newPath: String, newKey: Passphrase): SqlRow`
where the SqlRow returned is the results from the select? It'd be easy to do, I'll need to find or build an input unencrypted DB with at least one table in it for the unit test of the new function.
If this sounds useful, or you have something else in mind, lemme know here. Once we settle on a first draft we can add an issue and I can do the work pretty easily.
